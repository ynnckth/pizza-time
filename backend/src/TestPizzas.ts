import { Pizza } from './models/Pizza';

export const pizzaMargherita: Pizza = {
  name: 'Margherita',
  description: 'Plain and boring',
  unitPrice: 10,
  isAvailable: true,
};

export const pizzaSalami: Pizza = {
  name: 'Salami',
  description: 'Nice and spicy!',
  unitPrice: 12,
  isAvailable: true,
};
