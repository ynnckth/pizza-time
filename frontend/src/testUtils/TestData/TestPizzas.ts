import { OrderItem } from '../../../generated';

export const pizzaMargherita: OrderItem = {
  name: 'Margherita',
  description: 'Plain and boring',
  unitPrice: 10,
  available: true,
};

export const pizzaSalami: OrderItem = {
  name: 'Salami',
  description: 'Nice and spicy!',
  unitPrice: 12,
  available: true,
};
