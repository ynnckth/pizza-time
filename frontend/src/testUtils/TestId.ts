export enum TestId {
  APP_TITLE = 'APP_TITLE',
  LOADING_SPINNER = 'LOADING_SPINNER',

  // Marketplace
  MARKETPLACE_PIZZA_CARD = 'MARKETPLACE_PIZZA_CARD',
  MARKETPLACE_CART_BUTTON = 'MARKETPLACE_CART_BUTTON',
  MARKETPLACE_NO_OF_ORDER_ITEMS = 'MARKETPLACE_NO_OF_ORDER_ITEMS',
  MARKETPLACE_ADD_PIZZA_TO_CART = 'MARKETPLACE_ADD_PIZZA_TO_CART',

  // Checkout
  CHECKOUT_TOTAL_PRICE = 'CHECKOUT_TOTAL_PRICE',
  CHECKOUT_PLACE_ORDER_BUTTON = 'CHECKOUT_PLACE_ORDER_BUTTON',
  CHECKOUT_EMPTY_CART_MESSAGE = 'CHECKOUT_EMPTY_CART_MESSAGE',
  CHECKOUT_ORDER_ITEM = 'CHECKOUT_ORDER_ITEM',
  CHECKOUT_REMOVE_ORDER_ITEM = 'CHECKOUT_REMOVE_ORDER_ITEM',
  CHECKOUT_FORM_FIRST_NAME = 'CHECKOUT_FORM_FIRST_NAME',
  CHECKOUT_FORM_LAST_NAME = 'CHECKOUT_FORM_LAST_NAME',
  CHECKOUT_FORM_EMAIL = 'CHECKOUT_FORM_EMAIL',

  // Past orders
  PAST_ORDERS_ORDER = 'PAST_ORDERS_ORDER',
  PAST_ORDERS_NO_ORDERS_MESSAGE = 'PAST_ORDERS_NO_ORDERS_MESSAGE',
}
