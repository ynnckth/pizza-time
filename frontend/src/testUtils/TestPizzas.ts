import {OrderItem} from '../models/OrderItem';

export const pizzaMargherita: OrderItem = {
  name: 'Margherita',
  description: 'Plain and boring',
  unitPrice: 10,
  isAvailable: true,
};

export const pizzaSalami: OrderItem = {
  name: 'Salami',
  description: 'Nice and spicy!',
  unitPrice: 12,
  isAvailable: true,
};
