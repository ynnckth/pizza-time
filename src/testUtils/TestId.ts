export enum TestId {
  LOADING_SPINNER = 'LOADING_SPINNER',
  MARKETPLACE_PIZZA_CARD = 'MARKETPLACE_PIZZA_CARD',

  // Checkout
  CHECKOUT_TOTAL_PRICE = 'CHECKOUT_TOTAL_PRICE',
  CHECKOUT_PLACE_ORDER_BUTTON = 'CHECKOUT_PLACE_ORDER_BUTTON',
  CHECKOUT_EMPTY_CART_MESSAGE = 'CHECKOUT_EMPTY_CART_MESSAGE',
  CHECKOUT_ORDER_ITEM = 'CHECKOUT_ORDER_ITEM',
}
