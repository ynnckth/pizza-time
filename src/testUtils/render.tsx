import React from 'react';
import { render as rtlRender, RenderOptions } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

/**
 * Wraps the react-testing-library render function with other required providers
 * @param component
 * @param options
 */
export const render = (component: React.ReactElement, options?: Omit<RenderOptions, 'queries'>) => {
  return rtlRender(<BrowserRouter>{component}</BrowserRouter>, options);
};
