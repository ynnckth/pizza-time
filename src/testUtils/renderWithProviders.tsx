import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { PreloadedState } from '@reduxjs/toolkit';
import { AppStore, RootState, setupStore } from '../redux/Store';
import { Provider } from 'react-redux';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

/**
 * Wraps the react-testing-library render function with other required providers
 * like the redux store provider and a browser router.
 * @param component
 * @param preloadedState
 * @param store
 * @param renderOptions
 * For more information on testing with redux see:
 * https://redux.js.org/usage/writing-tests#setting-up-a-reusable-test-render-function
 */
export const renderWithProviders = (
  component: React.ReactElement,
  { preloadedState = {}, store = setupStore(preloadedState), ...renderOptions }: ExtendedRenderOptions = {}
) => {
  return render(
    <Provider store={store}>
      <BrowserRouter>{component}</BrowserRouter>
    </Provider>,
    renderOptions
  );
};
