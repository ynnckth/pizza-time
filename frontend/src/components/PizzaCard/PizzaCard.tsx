import React from 'react';
import { Button, Card, CardActions, CardContent, CardMedia, Typography } from '@mui/material';
import pizzaImage from '../../assets/pizza.png';
import { TestId } from '../../testUtils/TestId';
import { useAppDispatch } from '../../redux/Hooks';
import { addOrderItem } from '../../redux/Slices/Checkout/CheckoutSlice';
import { OrderItem } from '../../../generated';

interface Props {
  pizza: OrderItem;
}

const PizzaCard: React.FC<Props> = ({ pizza }) => {
  const dispatch = useAppDispatch();

  return (
    <Card sx={{ width: 250, height: 320, margin: 2 }} data-testid={TestId.MARKETPLACE_PIZZA_CARD}>
      <CardMedia component="img" height="140" src={pizzaImage} alt="delicious pizza" />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          {pizza.name}
        </Typography>
        <Typography variant="body2" color="text.secondary">
          {pizza.description}
        </Typography>
        <Typography variant={'body1'}>{`$${pizza.unitPrice}`}</Typography>
      </CardContent>
      <CardActions>
        <Button
          size="small"
          onClick={() => dispatch(addOrderItem(pizza))}
          data-testid={TestId.MARKETPLACE_ADD_PIZZA_TO_CART}
        >
          Add to cart
        </Button>
      </CardActions>
    </Card>
  );
};
export default PizzaCard;
