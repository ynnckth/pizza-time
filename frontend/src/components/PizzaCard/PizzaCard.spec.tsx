import PizzaCard from './PizzaCard';
import { pizzaMargherita } from '../../testUtils/TestPizzas';
import { renderWithProviders } from '../../testUtils/renderWithProviders';

describe('PizzaCard', () => {
  // Update snapshot with "jest --updateSnapshot"
  it('should match previous snapshot', () => {
    const { asFragment } = renderWithProviders(<PizzaCard pizza={pizzaMargherita} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
