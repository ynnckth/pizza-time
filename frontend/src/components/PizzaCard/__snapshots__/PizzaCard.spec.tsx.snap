// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PizzaCard Renders snapshot correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1peorf8-MuiPaper-root-MuiCard-root"
    data-testid="MARKETPLACE_PIZZA_CARD"
  >
    <img
      alt="delicious pizza"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="140"
      src="pizza.png"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Margherita
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        Plain and boring
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        $10
      </p>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
        data-testid="MARKETPLACE_ADD_PIZZA_TO_CART"
        tabindex="0"
        type="button"
      >
        Add to cart
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`PizzaCard should match previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1peorf8-MuiPaper-root-MuiCard-root"
    data-testid="MARKETPLACE_PIZZA_CARD"
  >
    <img
      alt="delicious pizza"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="140"
      src="pizza.png"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Margherita
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        Plain and boring
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        $10
      </p>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
        data-testid="MARKETPLACE_ADD_PIZZA_TO_CART"
        tabindex="0"
        type="button"
      >
        Add to cart
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
